src/DDTrace/Processing/TraceAnalyticsProcessor.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace DDTrace\Processing;
+
+use DDTrace\Span;
+use DDTrace\Tag;
+
+/**
+ * A span processor in charge of adding the trace analytics client config metric when appropriate.
+ */
+final class TraceAnalyticsProcessor
+{
+    /**
+     * Process the span adding the trace analytics client config option when appropriate.
+     *
+     * @param Span $span
+     */
+    public function process(Span $span)
+    {
+        // We only consider spans that are marked as trace analytics candidates, otherwise the customer bill would
+        // explode because we are sampling not relevant spans.
+        if (!$span->isTraceAnalyticsCandidate()) {
+            return;
+        }
+
+        // We only process spans that are generated by an integration, no custom span. Being part of an integration
+        // makes them configurable via the usual config options.
+        if (null === $span->getIntegration()) {
+            return;
+        }
+
+        // If a trace analytics tag has already been set, then we honor it.
+        if ($span->hasTag(Tag::ANALYTICS_KEY)) {
+            return;
+        }
+
+        $integration = $span->getIntegration();
+        if ($integration->isTraceAnalyticsEnabled()) {
```

Comment:
Since this the return value is nullable. :)

```suggestion
        if (null !== $integration && $integration->isTraceAnalyticsEnabled()) {
```

Reply:
Adding as a safety net, but we already check for the integration not being `null` on line 28 :)